front/src/common/rollingStockContext.tsx

```diff
@@ -0,0 +1,47 @@
+import { createContext, useContext, useMemo } from 'react';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { LightRollingStockWithLiveries } from 'common/api/osrdEditoastApi';
+
+type RollingStockContextValue = {
+  rollingStockMap: Map<string, LightRollingStockWithLiveries>;
+  isError: boolean;
+  rollingStocks: LightRollingStockWithLiveries[] | null;
+};
+
+const rollingStockContext = createContext<RollingStockContextValue | null>(null);
+
+export const useRollingStockContext = (): RollingStockContextValue => {
+  const context = useContext(rollingStockContext);
+  if (!context) {
+    throw new Error('useRollingStockContext must be used within a RollingStockContextProvider');
+  }
+  return context;
+};
+
+type RollingStockContextProviderProps = {
+  children: React.ReactNode;
+};
+
+export const RollingStockContextProvider = ({ children }: RollingStockContextProviderProps) => {
+  const { data, error } = osrdEditoastApi.endpoints.getLightRollingStock.useQuery({
+    pageSize: 1000,
+  });
+
+  const rollingStocksData = data?.results ?? [];
+
+  const rollingStockMap = useMemo(
+    () =>
+      new Map<string, LightRollingStockWithLiveries>(rollingStocksData.map((rs) => [rs.name, rs])),
+    [rollingStocksData]
+  );
+  const rollingStocks = useMemo(() => Array.from(rollingStockMap.values()), [rollingStockMap]);
+  const isError = Boolean(error);
+
+  const value = useMemo(
+    () => ({ rollingStockMap, isError, rollingStocks }),
+    [rollingStockMap, isError]
+  );
```

Comment:
- You can directly compute rollingStocksMap in this useMemo and remove the ones above
- Also I think rollingStocks is simply rollingStocksData no ?